src/models/Settings.js

```diff
@@ -238,6 +247,57 @@ function (Okta, Q, Errors, BrowserFeatures, Util, Logger, config) {
           return !!(socialAuthConfigured && idpDisplay.toUpperCase() === 'PRIMARY');
         },
         cache: true
+      },
+      configuredIdpsOptionsSet: {
+        deps: ['configuredSocialIdps'],
+        fn: function (configuredSocialIdps) {
+          _.each(configuredSocialIdps, function(idp) {
+            var type = idp.type.toLowerCase();
+            //set required properties
+            idp['attr'] = 'social-auth-' + type + '-button';
+            idp['className'] = 'social-auth-button ' + idp['attr'];
+            idp['title'] = Okta.loc('socialauth.' + type + '.label');
+            idp['click'] = socialAuthClickHandler;
+          })
+          return configuredSocialIdps;
+        },
+        cache: true
+      },
+
+      //custom buttons
+      customButtonsOptionsSet: {
+        deps: ['customButtons'],
+        fn: function (customButtons) {
+          _.each(customButtons, function(button) {
+            //add custom auth class name
+            button['className'] = button['className'] + ' default-custom-button';
```

Comment:
I think we should also get rid of any other properties that are sent to us. Basically use a whitelist of what we need. Something like
```
_.pick(button, 'click', 'className', 'title')
```